### PR TITLE
Local songs album covers

### PR DESCRIPTION
**Pull request by wong-justin**
### This is an addition to show cover images for local songs played through Spotify.

Spotify plays local songs and can incorporate them into playlists when the user adds them
(desktop Spotify -> Edit -> Preferences... -> Local Files -> Add A Source),
and when requested, the Spotify Web API will give main info like title and artist.
However, the cover image is not given (because it is not on Spotify servers).
[Example Local Song Spotify Response](https://developer.spotify.com/documentation/general/guides/local-files-spotify-playlists/)
The default backup image in /Resources will be shown in the widget.

This pull request amends that through a LocalSongManager class that reads audio file metadata in the local file directory. It requires the package TagLibSharp.

Local songs are probably a pretty small use case among users, but maybe someone else besides me would benefit too.
Hopefully the code style/structure fits the project since I've never contributed before.

Screenshots:

SettingsWithLocalSongsControls_blurred

LocalSongCoverDisplayed

### CODE REVIEW

File: Spotify Stream Widget/LocalSongManager.vb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    Public Sub New(dir As String)
+        InitLocalDir(dir)
+    End Sub
+
+    Public Sub InitLocalDir(dir As String)
+        'Given a directory path, set list of local song files (will searched thru later by Viewer)
+        'Should be async, but too much of a hassle. 
+        'Only UX annoyance will be that thumbnail won't update live when local dir is set.
+
+        For Each filepath In IO.Directory.GetFiles(dir, "*.*").Where(Function(f) IsAudio(f))
```
Comment (MarcoPNS) [inline Spotify Stream Widget/LocalSongManager.vb:22]: Does not work recursive. `System.IO.SearchOption.AllDirectories should fix` this.

File: Spotify Stream Widget/LocalSongManager.vb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                _dict.Add(key, filepath)
+
+            Catch ex As Exception
+                Log(2, "Couldn't add local song at " + filepath + ": " + ex.ToString())
+            End Try
+
+        Next
+
+        Log(1, "Added " + _dict.Count.ToString() + " local files: " + DictToString(_dict))
+    End Sub
```
Comment (MarcoPNS) [inline Spotify Stream Widget/LocalSongManager.vb:48]: Probably blocks the UI for a few seconds - would like to test it with a HUGE Folder. Should be async for final release.

File: Spotify Stream Widget/Viewer.vb [hunk 3]
```diff
@@ -300,7 +302,16 @@ Public Class Viewer
                 AlbumCover.Image = dummyImage
             End Try
         Else
-            AlbumCover.Image = My.Resources.albumArt
+            'Search local dir for song and read album cover image from file
+            Try
+
+                AlbumCover.Image = _localSongManager.ImageForSong(_playback.Item.Name, _playback.Item.Artists(0).Name)
```
Comment (MarcoPNS) [inline Spotify Stream Widget/Viewer.vb:308]: maybe check if the localSongManager is not nothing.

**Review by MarcoPNS — CHANGES_REQUESTED** [review]
First of all: Thank you for your PR! It's well written. I would like to do some testing and maybe switch to async before going live with this feature. Also it seems that you missed the `My.Settings` section.

**Comment by wong-justin** [comment]
Thanks for the feedback! I addressed the things you mentioned except for async. Anything else needing work before that?

**Pull request merged**